docs/plugins/filters/ruby.asciidoc

```diff
@@ -32,15 +38,100 @@ For example, to cancel 90% of events, you can do this:
       }
     }
 
-If you need to create additional events, it cannot be done as in other filters where you would use `yield`,
-you must use a specific syntax `new_event_block.call(event)` like in this example duplicating the input event
+If you need to create additional events, you must use a specific syntax `new_event_block.call(event)` like in this example duplicating the input event
 [source,ruby]
 filter {
   ruby {
     code => "new_event_block.call(event.clone)"
   }
 }
 
+===== Using a Ruby script file
+
+As the inline code can become complex and hard to structure inside of a text string in `code`, it's then preferrable to place the Ruby code in a .rb file, using the `path` option.
+
+[source,ruby]
+    filter {
+      ruby {
+        # Cancel 90% of events
+        path => "/etc/logstash/drop_percentage.rb"
+        script_params => { "percentage" => 0.9 }
+      }
+    }
+
+The ruby script file should define the following methods:
+
+ * `register(params)`: An optional register method that receives the key/value hash passed in the `script_params` configuration option
+ * `filter(event)`: A mandatory Ruby method that accepts a Logstash event and must return an array of events
+
+Below is an example implementation of the `drop_percentage.rb` ruby script that drops a configurable percentage of events:
+
+[source,ruby]
+----
+# the value of `params` is the value of the hash passed to `script_params`
+# in the logstash configuration
+def register(params)
+	@drop_percentage = params["percentage"]
+end
+
+# the filter method receives an event and must return a list of events.
+# Dropping an event means not including it in the return array,
+# while creating new ones only requires you to add a new instance of
+# LogStash::Event to the returned array
+def filter(event)
+	if rand >= @drop_percentage
+		return [event]
+	else
+		return [] # return empty array to cancel event
+	end
+end
+----
+
+====== Testing the ruby script
```

Comment:
Level 6 headings are not supported by the doc build. This renders as text. Change this to ===== Testing...